cairo/ethereum/cancun/vm/instructions/arithmetic.cairo

```diff
@@ -35,32 +35,35 @@ func add{
     with stack {
         let (x, err1) = pop();
         if (cast(err1, felt) != 0) {
+            EvmImpl.set_stack(stack);
             return err1;
         }
         let (y, err2) = pop();
         if (cast(err2, felt) != 0) {
+            EvmImpl.set_stack(stack);
             return err2;
         }
     }
 
     // GAS
     let err3 = charge_gas(Uint(GasConstants.GAS_VERY_LOW));
     if (cast(err3, felt) != 0) {
+        EvmImpl.set_stack(stack);
         return err3;
     }
 
     // OPERATION
     let (result, _) = uint256_add([x.value], [y.value]);
     with stack {
         let err4 = push(U256(new U256Struct(result.low, result.high)));
+        EvmImpl.set_stack(stack);
```

Comment:
why not use set_pc_and_stack like the other ?

Reply:
in error cases, PC is not incremented

Reply:
What I mean is that it is out of the error case and not updated at the end. Which is not like other test where it is rebinded in error case and at the end (which I find easier to follow)